src/snowflake/cli/_plugins/spcs/services/manager.py

```diff
@@ -199,6 +206,167 @@ def stream_logs(
         except KeyboardInterrupt:
             return
 
+    def get_account_event_table(self):
+        query = "show parameters like 'event_table' in account"
+        results = self.execute_query(query, cursor_class=DictCursor)
+        event_table = next(
+            (r["value"] for r in results if r["key"] == "EVENT_TABLE"), ""
+        )
+        if not event_table:
+            raise SPCSEventTableError("No SPCS event table configured in the account.")
+        return event_table
+
+    def get_events(
+        self,
+        service_name: str,
+        instance_id: str,
+        container_name: str,
+        since: str | datetime | None = None,
+        until: str | datetime | None = None,
+        first: int = -1,
```

Comment:
How about
```suggestion
        first: Optional[int] = None,
```
and later 
```
 first_clause = f"limit {first}" if first is not None else ""
 ```
 (and the same for "last")

Reply:
fixed.